src/main/scala/com/microsoft/azure/cosmosdb/spark/CosmosDBConnection.scala

```diff
@@ -22,281 +22,77 @@
-
-  @transient private var bulkImporter: DocumentBulkExecutor = _
-
-  private var documentClient: DocumentClient = CosmosDBConnection.getClient(connectionMode, getClientConfiguration(config))
-
-  logInfo(s"${timerName}: scheduling timer - delay: ${refreshDelay} ms, period: ${refreshPeriod} ms")
-  timer.schedule(
-    new TimerTask { def run() = {
-      collectionThroughput = None
-      logInfo(s"${timerName}: Resetting cached collection throughput")
-    }},
-    refreshDelay, 
-    refreshPeriod
-  )
-
-  def getPartitionKeyDefinition(partitionKeyDefinition: Option[String]) : PartitionKeyDefinition = {
-    if (partitionKeyDefinition.isDefined) {
-        val pkDefinition = new PartitionKeyDefinition()
-        val paths: ListBuffer[String] = new ListBuffer[String]()
-        paths.add(partitionKeyDefinition.get)
-        pkDefinition.setPaths(paths)  
-
-        pkDefinition
-    }
-    else {
-      getCollection.getPartitionKey
-    }
-  }
+  private val clientConfig = ClientConfiguration(config)
 
-  def getDocumentBulkImporter(collectionThroughput: Int, partitionKeyDefinition: Option[String], maxMiniBatchUpdateCount: Int, maxMiniBatchImportSizeKB: Int): DocumentBulkExecutor = {
-    if (bulkImporter == null) {
-      val initializationRetryOptions = new RetryOptions()
-      initializationRetryOptions.setMaxRetryAttemptsOnThrottledRequests(1000)
-      initializationRetryOptions.setMaxRetryWaitTimeInSeconds(1000)
-
-      val pkDefinition = getPartitionKeyDefinition(partitionKeyDefinition)
-
-      bulkImporter = DocumentBulkExecutor.builder.from(documentClient,
-        databaseName,
-        collectionName,
-        pkDefinition,
-        collectionThroughput
-      ).withInitializationRetryOptions(initializationRetryOptions)
-        .withMaxUpdateMiniBatchCount(maxMiniBatchUpdateCount)
-        .withMaxMiniBatchSize(maxMiniBatchImportSizeKB * 1024).build()
-    }
-
-    bulkImporter
-  }
-
-  def setDefaultClientRetryPolicy: Unit = {
-    if (documentClient != null) {
-      documentClient.getConnectionPolicy().getRetryOptions().setMaxRetryAttemptsOnThrottledRequests(1000);
-      documentClient.getConnectionPolicy().getRetryOptions().setMaxRetryWaitTimeInSeconds(1000);
-    }
-  }
-
-  def setZeroClientRetryPolicy: Unit = {
-    if (documentClient != null) {
-      documentClient.getConnectionPolicy().getRetryOptions().setMaxRetryAttemptsOnThrottledRequests(0);
-      documentClient.getConnectionPolicy().getRetryOptions().setMaxRetryWaitTimeInSeconds(0);
-    }
+  def getCollectionLink: String = {
+    collectionLink
   }
 
-  def getCollection: DocumentCollection = {
-    if (collection == null) {
-      collection = documentClient.readCollection(collectionLink, null).getResource
-    }
-    collection
-  }
-
-  def getDatabase: Database = {
-    if (database == null) {
-      database = documentClient.readDatabase(databaseLink, null).getResource
-    }
-    database
+  def reinitializeClient(): AnyVal = {
```

Comment:
This should return `Unit` (equivalent of `void`.  `AnyVal` is the equivalent of `object`)